website/addons/dataverse/static/dataverseNodeConfig.js

```diff
@@ -36,48 +41,73 @@ function ViewModel(url) {
     self.savedDataverseTitle = ko.observable();
     self.datasetWasFound = ko.observable(false);
 
+    self.accounts = ko.observable([]);
+    self.hosts = ko.observableArray([]);
+    self.selectedHost = ko.observable();    // Host specified in select element
+    self.customHost = ko.observable();      // Host specified in input element
+    self.savedHost = ko.observable();       // Configured host
+
+    // Designated host, specified from select or input element
+    self.host = ko.pureComputed(function() {
+        return self.useCustomHost() ? self.customHost() : self.selectedHost();
+    });
+    // Hosts visible in select element. Includes presets and "Other" option
+    self.visibleHosts = ko.pureComputed(function() {
+        return self.hosts().concat([otherString]);
+    });
+    // Whether to use select element or input element for host designation
+    self.useCustomHost = ko.pureComputed(function() {
+        return self.selectedHost() === otherString;
+    });
+    self.showApiTokenInput = ko.pureComputed(function() {
+        return typeof self.selectedHost() !== 'undefined';
```

Comment:
I used a `Boolean(self.selectedHost())`, since the value can now sometimes be `null`.